File: apus.client/src/app/activities/ActivityDto/ActivityDto.ts
export interface ActivityDto {
    id: string;
    title: string;
    description?: string;
    duration: string;        // or number, whatever you use
    date: string;
    avgHr?: number;
    totalCalories?: number;
    type: string;            // e.g. "RunningActivityDto", "HikingActivityDto", etc.
    displayname?: string;

    // gps fields (only present on some)
    distanceKm?: number;
    elevationGain?: number;
    pace?: number;           // only on running

    likescount?: number;
}

export interface DisplayProp {
    label: string;
    value: string | number;
}

File: APUS.Server/DTOs/ActivityDto.cs
namespace APUS.Server.DTOs
{
	public record ActivityDto
	{
		public string Id { get; set; }
		public string Title { get; set; }
		public string? Description { get; set; }
		public TimeSpan Duration { get; set; }
		public DateTime Date {  get; set; }
		public int? AvgHr { get; set; }
		public int? TotalCalories { get; set; }
		public string Type { get; set; }
		public string? DisplayName { get; set; }
		public int? LikesCount { get; set; }
	}

	public record GpsActivityDto : ActivityDto
	{
		public double? DistanceKm { get; set; }
		public double? ElevationGain { get; set; }
	}

	public record RunningActivityDto : GpsActivityDto
	{
		public double? Pace { get; set; }
	}
}
